### DISCUSSION TITLE
Integrating Python-Based Nada Package with a JavaScript Framework for the Nada Playground Web App

### DISCUSSION THREAD

**Discussion by yashmangal112** [NillionNetwork/discussions#96] [General]
I am working on building the Nada Playground Web App, but I'm facing an issue because the Nada package is only available for Python and not as an NPM package. This might mean I need to incorporate Python into my project. But for that project, I should use JavaScript Framework.  What steps should I take to resolve this?

**Comment by jcabrero** [comment]
Hi! Thanks for commenting. There exists already a sandbox for users to play with called [Nada Sandbox](https://nillion.pub/nada-sandbox/). In any case, if you chose to do a similar project, you could profit from Javascript bridging projects like `PyScript` to run Python in your browser.

**Reply by yashmangal112** [reply]
Yeah! I discovered the existing Nada Sandbox. As part of Builder Bounty Submission projects, I'm developing a Nada Playground using a JavaScript framework. That is why I want to use some scripts like ```jsnadac``` script and ```nada_run``` script from Nada Sandbox for my project. So, can you please tell me how I can use these libraries for my JavaScript framework? 
Thanks

**Reply by lapets** [reply]
It's great to hear you're interested in this project! As mentioned above, the [Nada Sandbox source code](https://github.com/NillionNetwork/nada-sandbox) demonstrates two approaches that can be used to simulate Nada programs.

* One place you could begin is by putting together a few [PyScript](https://pyscript.net/) examples on their own to learn how you can run Python code in a browser that's invoked using your JavaScript code. You can check out their [examples](https://docs.pyscript.net/2024.8.2/examples/) to learn more. Next, go over the Nada and Nillion tutorials to learn how you can use the [nada-dsl](https://pypi.org/project/nada-dsl/) library. Finally, you can put these two together in a way that resembles the Nada Sandbox implementation. For example, you can take a look at how [Nada Sandbox runs Python code](https://github.com/NillionNetwork/nada-sandbox/blob/c116bae00c7f4423d431a6b894a4f8c38611cb67/index.html#L228) to simulate a Nada program.

* Alternatively, you can take a look at [how Nada Sandbox invokes `jsnadac`](https://github.com/NillionNetwork/nada-sandbox/blob/c116bae00c7f4423d431a6b894a4f8c38611cb67/jsnadac.html). This relies on a compiled WebAssembly Nada simulator. 

For either approach, the goal is to use a contemporary JavaScript UI framework such as React in order to build a user-friendly and maintainable app. We hope this helps!
